### ISSUE TITLE
conda-build tries to compile .pyc files in non-python build

### ISSUE THREAD

**Issue by mwiebe** [conda/conda-build#1277]
Python is not a dependency of the recipe triggering this. One thing different about this recipe than most is that it includes `binary_relocation: False`

Versions of conda:

```
docker@mark-wiebe-docker-wilypy35:~$ conda list | grep conda
# packages in environment at /home/docker/miniconda3:
conda                     4.1.11                   py35_0
conda-build               1.21.14                  py35_0
conda-env                 2.5.2                    py35_0
```

The error:

```
2016-08-26 02:59:41:  0: STDOUT: compiling .pyc files...
2016-08-26 02:59:41:  0: STDOUT: Traceback (most recent call last):
2016-08-26 02:59:41:  0: STDOUT:   File "/home/docker/miniconda3/bin/conda-build", line 5, in <module>
2016-08-26 02:59:41:  0: STDOUT:     sys.exit(main())
2016-08-26 02:59:41:  0: STDOUT:   File "/home/docker/miniconda3/lib/python3.5/site-packages/conda_build/main_build.py", line 151, in main
2016-08-26 02:59:41:  0: STDOUT:     args_func(args, p)
2016-08-26 02:59:41:  0: STDOUT:   File "/home/docker/miniconda3/lib/python3.5/site-packages/conda_build/main_build.py", line 412, in args_func
2016-08-26 02:59:41:  0: STDOUT:     args.func(args, p)
2016-08-26 02:59:41:  0: STDOUT:   File "/home/docker/miniconda3/lib/python3.5/site-packages/conda_build/main_build.py", line 355, in execute
2016-08-26 02:59:41:  0: STDOUT:     debug=args.debug)
2016-08-26 02:59:41:  0: STDOUT:   File "/home/docker/miniconda3/lib/python3.5/site-packages/conda_build/build.py", line 650, in build
2016-08-26 02:59:41:  0: STDOUT:     preserve_egg_dir=bool(m.get_value('build/preserve_egg_dir')))
2016-08-26 02:59:41:  0: STDOUT:   File "/home/docker/miniconda3/lib/python3.5/site-packages/conda_build/post.py", line 194, in post_process
2016-08-26 02:59:41:  0: STDOUT:     compile_missing_pyc(files)
2016-08-26 02:59:41:  0: STDOUT:   File "/home/docker/miniconda3/lib/python3.5/site-packages/conda_build/post.py", line 189, in compile_missing_pyc
2016-08-26 02:59:41:  0: STDOUT:     call([python_exe, '-Wi', '-m', 'py_compile', f], cwd=cwd)
2016-08-26 02:59:41:  0: STDOUT:   File "/home/docker/miniconda3/lib/python3.5/subprocess.py", line 557, in call
2016-08-26 02:59:41:  0: STDOUT:     with Popen(*popenargs, **kwargs) as p:
2016-08-26 02:59:41:  0: STDOUT:   File "/home/docker/miniconda3/lib/python3.5/subprocess.py", line 947, in __init__
2016-08-26 02:59:41:  0: STDOUT:     restore_signals, start_new_session)
2016-08-26 02:59:41:  0: STDOUT:   File "/home/docker/miniconda3/lib/python3.5/subprocess.py", line 1551, in _execute_child
2016-08-26 02:59:41:  0: STDOUT:     raise child_exception_type(errno_num, err_msg)
2016-08-26 02:59:41:  0: STDOUT: FileNotFoundError: [Errno 2] No such file or directory: '/home/docker/miniconda3/envs/_build_placehold_placehold_placehold_placehold_plac/bin/python'
```

**Comment by msarahan** [comment]
Thanks for raising this.  It is addressed by https://github.com/conda/conda-build/pull/1261 - and will be a part of the conda-build 2.0 release, slated for Monday.

**Comment by mwiebe** [comment]
Sweet, a little timing coincidence here. :)

**Comment by msarahan** [comment]
This is released in the 2.0.0beta4 release, which is available on the conda-canary channel.  If you have time to test, I'd greatly value your opinions and reports of any oddities.

I'm closing this issue for now, but please reopen or open anew as necessary.

**Comment by mwiebe** [comment]
@msarahan the job which failed now succeeded on the canary channel, thanks!
